### PR TITLE
Backend training complete

### PR DESCRIPTION
**Pull request by leonardtang**
Unsure if GetUserInfo() / GetUserInfoService() is supposed to require password as a parameter? As such, didn't include as many integration tests (getting user info) as unit tests (login). Otherwise, things should be good. 

Was being silly earlier and accidentally made PR to main

### CODE REVIEW

File: src/test/User/LoginServiceUnitTests.java [hunk 1]
```diff
@@ -37,8 +37,41 @@ public void userNotFound() {
...
+    EntityFactory.createUser().withUsername("username1").withPasswordToHash("abcefghijk").buildAndPersist(userDao);
+    LoginService loginService = new LoginService(userDao, logger, "username1", "abcefghijk");
+    Message message = loginService.executeAndGetResponse();
+    assertEquals(UserMessage.SUCCESS, message);
+  }
+
+  @Test
+  public void userFoundWrongPassword() {
+    EntityFactory.createUser().withUsername("username1").withPasswordToHash("abc").buildAndPersist(userDao);
+    LoginService loginService = new LoginService(userDao, logger, "username1", "123");
```
Comment (nrod80) [inline src/test/User/LoginServiceUnitTests.java:56]: We should probably test with an incorrect password that is >= 8 characters to verify that auth fails when comparing the hash of an incorrect password. With a short password, I believe this only tests is the `ValidationUtils.isValidPassword` logic.

**Review by nrod80 — COMMENTED** [review]
`GetUserInfo` doesn't require a password, since we [determine the `username` based on the session](https://github.com/keepid/training_server_stripped/blob/fb6dcf07672070796feb2909f07ad8f38b8b1a97/src/main/User/UserController.java#L44), which implies that the user is already authenticated.

It would be awesome if you could add a few integration tests too, but otherwise this looks great to me! Nice work 👍

**Review by crchong1 — APPROVED** [review]
This looks great- i like the different tests. I would maybe in the future add one or two more edge cases, but other than that this is fine
